updateStableIds/src/main/java/org/reactome/release/update_stable_ids/UpdateStableIds.java

```diff
@@ -0,0 +1,170 @@
+package org.reactome.release.update_stable_ids;
+
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+import java.util.stream.Collector;
+import java.util.stream.Collectors;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.gk.model.GKInstance;
+import org.gk.model.ReactomeJavaConstants;
+
+import static org.gk.model.ReactomeJavaConstants.*;
+import org.gk.persistence.MySQLAdaptor;
+import org.gk.persistence.TransactionsNotSupportedException;
+import org.gk.schema.GKSchemaClass;
+import org.gk.schema.InvalidAttributeException;
+import org.reactome.release.common.database.InstanceEditUtils;
+
+public class UpdateStableIds {
+
+	private static final Logger logger = LogManager.getLogger();
+	
+	public static void stableIdUpdater(MySQLAdaptor dbaSlice, MySQLAdaptor dbaPrevSlice, MySQLAdaptor dbaGkCentral, Long personId) throws Exception {
+		
+		logger.info("Generating InstanceEdits for " + dbaSlice.getDBName() + " and " + dbaGkCentral.getDBName());
+		// Instance Edits for test_slice and gk_central
+		String creatorName = "org.reactome.release.update_stable_ids";
+		GKInstance sliceIE = InstanceEditUtils.createInstanceEdit(dbaSlice, personId, creatorName);
+		GKInstance gkCentralIE = InstanceEditUtils.createInstanceEdit(dbaGkCentral, personId, creatorName);
+
+		// At time of writing (December 2018), test_slice is a non-transactional database. This check has been put in place as a safety net in case that changes.
+		if (dbaSlice.supportsTransactions()) {
+			dbaSlice.startTransaction();
+		}
+		dbaGkCentral.startTransaction();
+		
+		//TODO:Should legacy regulation instances be updated?
```

Comment:
Should this question be answered/resolved before submitting a pull-request, or is this a longer-term issue?

Reply:
This one is longer-term, I put it in the code so that it wasn't forgotten

Reply:
It seems like for now, we are not concerned about stable identifiers for regulation instances.  The idea of having more classes receive stable identifiers was mentioned, though (e.g. catalyst activities and possibly regulation instances again).  So, this is something to watch but nothing to do right now.